lib/statemanager.js

```diff
@@ -915,14 +916,21 @@ StateManager.prototype.isUnlocked = function(address) {
   return this.unlocked_accounts[address.toLowerCase()] != null;
 };
 
-StateManager.prototype.createFakeTransactionWithCorrectNonce = function(rawTx, from, callback) {
+StateManager.prototype.createTransactionWithCorrectNonce = function(rawTx, from, callback) {
   const self = this;
   self.blockchain.getQueuedNonce(from, (err, expectedNonce) => {
     if (err) {
       return callback(err);
     }
 
-    var tx = new FakeTransaction(rawTx);
+    var tx;
+    if (rawTx.hasOwnProperty("v") && rawTx.hasOwnProperty("r") && rawTx.hasOwnProperty("s")) {
```

Comment:
I like the idea (especially if we refactor) but im not sure it makes sense here/now.  Beyond this point, FakeTransaction/Transactions come out of the db as rawTx's which gives us no way (to my knowledge) to differentiate.  The signature parameters are populated in both cases and tx.verifySignature() is true for both tx types (if i later instantiate with either Fake/Transaction) regardless of whether the tx is 'eth_sendRaw..' or not.

Reply:
I like the idea (especially for a refactor) but im not sure makes sense here/now.  I'll explore this and report back.

Reply:
Checkout branch `tx-hash-collision-refactor`, it was done quickly but winds up breaking tests across the suite.  This seems like it will require some nuance that I think would be counterproductive to solve at this stage.  This should be in a project-wide refactor though - without question.